test/apps/iodemo/io_demo.cc

```diff
@@ -948,6 +948,21 @@ class P2pDemoCommon : public UcxContext {
         validate(conn, msg, iomsg_size);
     }
 
+    static inline void validate_msg(const iomsg_t *msg, UcxConnection *conn)
```

Comment:
can it be part of validate now?

Reply:
validate() is a const method, this one changes connection state. IMO better to keep it separately

Reply:
then maybe no need to set the id in validate method and just do this once as it was in the original code.
Having validate method which changes the state is not transparent even if this is done in a separate method